From 9bb4ee8e3c6909d6b6ff3267f31ca8f537f96984 Mon Sep 17 00:00:00 2001
From: David Lechner <david@pybricks.com>
Date: Thu, 11 Jun 2020 16:41:22 +0000
Subject: [bluetooth] fix compile error in dbus fake bluetooth device client

This fixes a compile error that was accidentally introduced in
https://crrev.com/2191232. For unknown reasons this does not cause a
compile error with libc++, which is normally used with chromium, but
does cause an error with stdlibc++.

The API for preparing write requests was not changed in
https://crrev.com/2191232, so the change that introduced this error
was incorrect anyway. This reverts the change that introduced the
error and adds the kTypeRequest argument to the call to WriteValue()
since that was the API that was actually changed in the prior CL.

Bug: 1092470
Change-Id: I6f6fe90ab42aea3db8cac95b69aee5095ea33b31
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2237388
Reviewed-by: Sonny Sasaka <sonnysasaka@chromium.org>
Commit-Queue: David Lechner <david@pybricks.com>
Cr-Commit-Position: refs/heads/master@{#777365}
---
 device/bluetooth/dbus/fake_bluetooth_device_client.cc | 7 +++----
 device/bluetooth/dbus/fake_bluetooth_device_client.h  | 2 +-
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/device/bluetooth/dbus/fake_bluetooth_device_client.cc b/device/bluetooth/dbus/fake_bluetooth_device_client.cc
index f7136afcb25f..23c5fe33b0bc 100644
--- a/device/bluetooth/dbus/fake_bluetooth_device_client.cc
+++ b/device/bluetooth/dbus/fake_bluetooth_device_client.cc
@@ -649,10 +649,9 @@ void FakeBluetoothDeviceClient::ExecuteWrite(
   for (const auto& prepare_write_request : prepare_write_requests_) {
     bluez::BluezDBusManager::Get()
         ->GetBluetoothGattCharacteristicClient()
-        ->WriteValue(std::get<0>(prepare_write_request),
-                     std::get<1>(prepare_write_request),
-                     std::get<2>(prepare_write_request), base::DoNothing(),
-                     base::DoNothing());
+        ->WriteValue(prepare_write_request.first, prepare_write_request.second,
+                     bluetooth_gatt_characteristic::kTypeRequest,
+                     base::DoNothing(), base::DoNothing());
   }
   prepare_write_requests_.clear();
   std::move(callback).Run();
diff --git a/device/bluetooth/dbus/fake_bluetooth_device_client.h b/device/bluetooth/dbus/fake_bluetooth_device_client.h
index b46715b5a665..98a9676382a2 100644
--- a/device/bluetooth/dbus/fake_bluetooth_device_client.h
+++ b/device/bluetooth/dbus/fake_bluetooth_device_client.h
@@ -384,7 +384,7 @@ class DEVICE_BLUETOOTH_EXPORT FakeBluetoothDeviceClient
   bool delay_start_discovery_;
 
   // Pending prepare write requests.
-  std::vector<std::tuple<dbus::ObjectPath, std::vector<uint8_t>, std::string>>
+  std::vector<std::pair<dbus::ObjectPath, std::vector<uint8_t>>>
       prepare_write_requests_;
 
   bool should_leave_connections_pending_;
-- 
2.20.1

